### PR TITLE
clear calcdata before redoing it in animations

### PR DESCRIPTION
**Pull request by alexcjohnson**
Fixes the issue we're seeing with calcTransforms + animations, which was introduced in #1717 
Then the question is: what do we test this with and how?

cc @etpinard @rreusser

### CODE REVIEW

File: src/plots/plots.js [hunk 1]
```diff
@@ -1813,6 +1813,10 @@ plots.transition = function(gd, data, layout, traces, frameOpts, transitionOpts)
         // of essentially the whole supplyDefaults step, so that it seems sensible to just use
         // supplyDefaults even though it's heavier than would otherwise be desired for
         // transitions:
+
+        // first delete calcdata so supplyDefaults knows a calc step is coming
+        delete gd.calcdata;
```
Comment (etpinard) [inline src/plots/plots.js]: Looks like a solid fix. It will slow down `Plotly.animate` though.

Comment (etpinard) [inline src/plots/plots.js]: There's some [logic](https://github.com/plotly/plotly.js/blob/6b7c9d7aeba575c3ff5a35a03e111c051e001ed7/src/plots/plots.js#L2026) in `Plots.doCalcdata` involving old versus new calcdata that might be obsolete now.

Comment (alexcjohnson) [inline src/plots/plots.js]: will it? We're calling `doCalcdata` no matter what so I'd think this will actually speed it up a tiny bit!

Comment (etpinard) [inline src/plots/plots.js]: Good point :+1:

**Comment by rreusser** [comment]
Here's at least a codepen that isolates this effect (plotly-latest works though so it's not visible on the codepen)

https://codepen.io/rsreusser/pen/rmbrXg?editors=0010

**Comment by etpinard** [comment]
Great thanks!

**Pull request merged**